source/components/Avatar/index.js

```diff
@@ -7,20 +7,31 @@ import { getAvatarImage } from './utils';
 
 import './styles.scss';
 
-const Avatar = ({
-    alt,
-    badge,
-    className,
-    href,
-    src,
-    title,
-    linkBuilder,
-}) => (
-    <div className={classNames('wds-avatar', className)} title={title}>
-        {getAvatarImage(href, alt, src, linkBuilder)}
-        {badge && <Badge badge={badge} />}
-    </div>
-);
+class Avatar extends React.Component {
```

Comment:
Can we use a functional component instead? Not sure what @vforge thinks but I would say everything should be functional components now that we have hooks. The _only_ case is for ErrorBoundaries with `componentDidCatch`

Reply:
Not sure how I could use a functional component when it needs a lifecycle method.

Reply:
You can useEffect for that ;>